src/tools/rbd_mirror/InstanceWatcher.cc

```diff
@@ -166,12 +272,57 @@ void InstanceWatcher<I>::remove(Context *on_finish) {
 }
 
 template <typename I>
-void InstanceWatcher<I>::handle_notify(uint64_t notify_id, uint64_t handle,
-                                       uint64_t notifier_id, bufferlist &bl) {
-  dout(20) << dendl;
+void InstanceWatcher<I>::notify_image_acquire(
+  const std::string &instance_id, const std::string &global_image_id,
+  const std::string &peer_mirror_uuid, const std::string &peer_image_id,
+  Context *on_notify_ack) {
+  dout(20) << "instance_id=" << instance_id << ", global_image_id="
+           << global_image_id << dendl;
+
+  Mutex::Locker locker(m_lock);
 
-  bufferlist out;
-  acknowledge_notify(notify_id, handle, out);
+  assert(m_on_finish == nullptr);
+
+  if (instance_id == m_instance_id) {
+    handle_image_acquire(global_image_id, peer_mirror_uuid, peer_image_id,
+                         on_notify_ack);
+  } else {
+    uint64_t request_id = ++m_request_seq;
+    bufferlist bl;
+    ::encode(NotifyMessage{ImageAcquirePayload{request_id, global_image_id,
+        peer_mirror_uuid, peer_image_id}}, bl);
```

Comment:
@trociny yeah, preferably it should be indented to the same level as its associated paramters (```request_id``` in this case) or move all the parameters down to a new (indented) line.